File: RepoTransBench/target_projects/C/Python/codeplea_genann/public_tests/test_public_preset_read.py
import pytest
import numpy as np
from src.genann import *

def test_public_preset_read():
    import os
    save_name = "example/xor.ann"
    if not os.path.exists(save_name):
        pytest.skip(f"Couldn't open file: {save_name}")

    with open(save_name, "rb") as saved:
        ann = genann_read(saved)
    assert ann is not None

    input_data = [[0,0],[0,1],[1,0],[1,1]]
    print("NAND test cases with loaded ANN:")
    for i in range(4):
        print(f"Output for {input_data[i]} is {genann_run(ann, input_data[i])}.")
    genann_free(ann)

File: RepoTransBench/target_projects/C/Rust/codeplea_genann/public_tests/test_public_preset_read.rs
use codeplea_genann_rs::*;
use std::fs::File;
use std::io::{Write, Read, BufReader};
use std::path::Path;

#[test]
fn public_preset_read() {
    // Expects "example/xor.ann" exists as in C example
    let save_name = "example/xor.ann";
    let exists = Path::new(save_name).exists();
    if !exists {
        eprintln!("Couldn't open file: {}", save_name);
        // skip this test if file does not exist
        return;
    }
    let mut saved = File::open(save_name).unwrap();
    let ann = genann_read(&mut saved);
    if ann.is_none() {
        panic!("Error loading ANN from file: {}.", save_name);
    }
    let mut ann = ann.unwrap();

    let input: [[f64; 2]; 4] = [
        [0.0, 0.0],
        [0.0, 1.0],
        [1.0, 0.0],
        [1.0, 1.0],
    ];
    for i in 0..4 {
        println!(
            "Output for [{}, {}] is {}.",
            input[i][0], input[i][1], ann.run(Some(&input[i]))[0]
        );
    }
}